tools/migrate_logs.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3.9
+import asyncio
+import re
+
+import databases
+
+# config (touch this)
+# mysql://{username}:{passwd}@{host}:{port}/{database}
+DB_DSN = "mysql://cmyui:lol123@localhost:3306/gulag_old"
+LOG_REGEX = re.compile(
+    r"<(.*)\((.*)\)> (?P<action>unrestricted|restricted|unsilenced|silenced|added note) ?(\((.*)\))? ?(\: (?P<note>.*))? ?(?:for (?P<reason>.*))?",
+)
+
+
+async def main() -> int:
+    async with databases.Database(DB_DSN) as db:
+        async with (
+            db.connection() as select_conn,
+            db.connection() as update_conn,
+        ):
+            # add/adjust new columns, keeping them null until we are finished
+            print("Creating new columns")
+
+            await update_conn.execute(
+                "ALTER TABLE `logs` ADD COLUMN `action` VARCHAR(32) null after `to`",
+            )
+            await update_conn.execute(
+                "ALTER TABLE `logs` MODIFY `msg` VARCHAR(2048) null",
+            )  # now used as reason
+
+            # get all logs & change
+            print("Getting all old logs")
+            for row in await select_conn.fetch_all(f"SELECT * FROM logs"):
+                note = row["msg"]
+
+                note_match = LOG_REGEX.match(row["msg"])
+                if not note_match:
+                    continue
```

Comment:
perhaps we should alert the user that not all logs will be preserved? they may want to back them up if important

Reply:
we could also save to a file (or add an option to do so)

Reply:
this is going to keep them in the db, with their old format: for that reason. i made the decision cus jkbgl for example tested the branch pre-migrator and i didn't want his already converted notes to be fucked by it

Reply:
ok yep awesome works for me